### PR TITLE
python310Packages.marshmallow: 3.13.0 -> 3.15.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
https://marshmallow.readthedocs.io/en/latest/changelog.html#id1

Fix build (https://hydra.nixos.org/build/177044261)

ZHF: #172160

CC: https://github.com/orgs/NixOS/teams/nixos-release-managers

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 173429` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>python310Packages.apache-airflow</li>
    <li>python310Packages.hug</li>
    <li>python39Packages.apache-airflow</li>
    <li>python39Packages.hug</li>
  </ul>
</details>
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>ansible-doctor (python39Packages.ansible-doctor)</li>
    <li>flashfocus</li>
    <li>python310Packages.ansible-doctor</li>
    <li>python310Packages.flask-appbuilder</li>
    <li>python310Packages.flask_marshmallow</li>
    <li>python310Packages.gradient</li>
    <li>python310Packages.marshmallow-oneofschema</li>
    <li>python310Packages.spyse-python</li>
    <li>python39Packages.flask-appbuilder</li>
    <li>python39Packages.flask_marshmallow</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.marshmallow-oneofschema</li>
    <li>sublime-music</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>esphome</li>
    <li>faraday-agent-dispatcher</li>
    <li>home-assistant-component-tests.renault</li>
    <li>home-assistant-component-tests.sia</li>
    <li>json-schema-for-humans (python39Packages.json-schema-for-humans)</li>
    <li>platformio</li>
    <li>python310Packages.aiohttp-apispec</li>
    <li>python310Packages.apispec</li>
    <li>python310Packages.dataclasses-json</li>
    <li>python310Packages.deezer-python</li>
    <li>python310Packages.environs</li>
    <li>python310Packages.faraday-agent-parameters-types</li>
    <li>python310Packages.json-schema-for-humans</li>
    <li>python310Packages.lektor</li>
    <li>python310Packages.marshmallow</li>
    <li>python310Packages.marshmallow-dataclass</li>
    <li>python310Packages.marshmallow-enum</li>
    <li>python310Packages.marshmallow-polyfield</li>
    <li>python310Packages.marshmallow-sqlalchemy</li>
    <li>python310Packages.pysiaalarm</li>
    <li>python310Packages.pytenable</li>
    <li>python310Packages.renault-api</li>
    <li>python310Packages.webargs</li>
    <li>python39Packages.aiohttp-apispec</li>
    <li>python39Packages.apispec</li>
    <li>python39Packages.dataclasses-json</li>
    <li>python39Packages.deezer-python</li>
    <li>python39Packages.environs</li>
    <li>python39Packages.faraday-agent-parameters-types</li>
    <li>python39Packages.lektor</li>
    <li>python39Packages.marshmallow</li>
    <li>python39Packages.marshmallow-dataclass</li>
    <li>python39Packages.marshmallow-enum</li>
    <li>python39Packages.marshmallow-polyfield</li>
    <li>python39Packages.marshmallow-sqlalchemy</li>
    <li>python39Packages.pysiaalarm</li>
    <li>python39Packages.pytenable</li>
    <li>python39Packages.renault-api</li>
    <li>python39Packages.webargs</li>
    <li>tribler</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 173429` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>python310Packages.apache-airflow</li>
    <li>python310Packages.hug</li>
    <li>python39Packages.apache-airflow</li>
    <li>python39Packages.hug</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>sublime-music</li>
  </ul>
</details>
<details>
  <summary>52 packages built:</summary>
  <ul>
    <li>ansible-doctor (python39Packages.ansible-doctor)</li>
    <li>esphome</li>
    <li>faraday-agent-dispatcher</li>
    <li>flashfocus</li>
    <li>home-assistant-component-tests.renault</li>
    <li>home-assistant-component-tests.sia</li>
    <li>json-schema-for-humans (python39Packages.json-schema-for-humans)</li>
    <li>platformio</li>
    <li>python310Packages.aiohttp-apispec</li>
    <li>python310Packages.ansible-doctor</li>
    <li>python310Packages.apispec</li>
    <li>python310Packages.dataclasses-json</li>
    <li>python310Packages.deezer-python</li>
    <li>python310Packages.environs</li>
    <li>python310Packages.faraday-agent-parameters-types</li>
    <li>python310Packages.flask-appbuilder</li>
    <li>python310Packages.flask_marshmallow</li>
    <li>python310Packages.gradient</li>
    <li>python310Packages.json-schema-for-humans</li>
    <li>python310Packages.lektor</li>
    <li>python310Packages.marshmallow</li>
    <li>python310Packages.marshmallow-dataclass</li>
    <li>python310Packages.marshmallow-enum</li>
    <li>python310Packages.marshmallow-oneofschema</li>
    <li>python310Packages.marshmallow-polyfield</li>
    <li>python310Packages.marshmallow-sqlalchemy</li>
    <li>python310Packages.pysiaalarm</li>
    <li>python310Packages.pytenable</li>
    <li>python310Packages.renault-api</li>
    <li>python310Packages.spyse-python</li>
    <li>python310Packages.webargs</li>
    <li>python39Packages.aiohttp-apispec</li>
    <li>python39Packages.apispec</li>
    <li>python39Packages.dataclasses-json</li>
    <li>python39Packages.deezer-python</li>
    <li>python39Packages.environs</li>
    <li>python39Packages.faraday-agent-parameters-types</li>
    <li>python39Packages.flask-appbuilder</li>
    <li>python39Packages.flask_marshmallow</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.lektor</li>
    <li>python39Packages.marshmallow</li>
    <li>python39Packages.marshmallow-dataclass</li>
    <li>python39Packages.marshmallow-enum</li>
    <li>python39Packages.marshmallow-oneofschema</li>
    <li>python39Packages.marshmallow-polyfield</li>
    <li>python39Packages.marshmallow-sqlalchemy</li>
    <li>python39Packages.pysiaalarm</li>
    <li>python39Packages.pytenable</li>
    <li>python39Packages.renault-api</li>
    <li>python39Packages.webargs</li>
    <li>tribler</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 173429` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>python310Packages.apache-airflow</li>
    <li>python310Packages.hug</li>
    <li>python310Packages.pytest-ansible</li>
    <li>python39Packages.apache-airflow</li>
    <li>python39Packages.hug</li>
    <li>python39Packages.pytest-ansible</li>
  </ul>
</details>
<details>
  <summary>78 packages built:</summary>
  <ul>
    <li>ansible (python39Packages.ansible-core)</li>
    <li>ansible-doctor (python39Packages.ansible-doctor)</li>
    <li>ansible-lint (python39Packages.ansible-lint)</li>
    <li>appdaemon</li>
    <li>deepdiff (python39Packages.deepdiff)</li>
    <li>esphome</li>
    <li>faraday-agent-dispatcher</li>
    <li>flashfocus</li>
    <li>home-assistant-component-tests.apple_tv</li>
    <li>home-assistant-component-tests.renault</li>
    <li>home-assistant-component-tests.sia</li>
    <li>json-schema-for-humans (python39Packages.json-schema-for-humans)</li>
    <li>kargo</li>
    <li>platformio</li>
    <li>python310Packages.aiohttp-apispec</li>
    <li>python310Packages.ansible</li>
    <li>python310Packages.ansible-compat</li>
    <li>python310Packages.ansible-core</li>
    <li>python310Packages.ansible-doctor</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-lint</li>
    <li>python310Packages.ansible-runner</li>
    <li>python310Packages.apispec</li>
    <li>python310Packages.dataclasses-json</li>
    <li>python310Packages.deepdiff</li>
    <li>python310Packages.deezer-python</li>
    <li>python310Packages.environs</li>
    <li>python310Packages.faraday-agent-parameters-types</li>
    <li>python310Packages.flask-appbuilder</li>
    <li>python310Packages.flask_marshmallow</li>
    <li>python310Packages.gradient</li>
    <li>python310Packages.json-schema-for-humans</li>
    <li>python310Packages.lektor</li>
    <li>python310Packages.marshmallow</li>
    <li>python310Packages.marshmallow-dataclass</li>
    <li>python310Packages.marshmallow-enum</li>
    <li>python310Packages.marshmallow-oneofschema</li>
    <li>python310Packages.marshmallow-polyfield</li>
    <li>python310Packages.marshmallow-sqlalchemy</li>
    <li>python310Packages.pyatv</li>
    <li>python310Packages.pysiaalarm</li>
    <li>python310Packages.pytenable</li>
    <li>python310Packages.renault-api</li>
    <li>python310Packages.runway-python</li>
    <li>python310Packages.spyse-python</li>
    <li>python310Packages.ttp</li>
    <li>python310Packages.ttp-templates</li>
    <li>python310Packages.webargs</li>
    <li>python39Packages.aiohttp-apispec</li>
    <li>python39Packages.ansible</li>
    <li>python39Packages.ansible-compat</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.ansible-runner</li>
    <li>python39Packages.apispec</li>
    <li>python39Packages.dataclasses-json</li>
    <li>python39Packages.deezer-python</li>
    <li>python39Packages.environs</li>
    <li>python39Packages.faraday-agent-parameters-types</li>
    <li>python39Packages.flask-appbuilder</li>
    <li>python39Packages.flask_marshmallow</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.lektor</li>
    <li>python39Packages.marshmallow</li>
    <li>python39Packages.marshmallow-dataclass</li>
    <li>python39Packages.marshmallow-enum</li>
    <li>python39Packages.marshmallow-oneofschema</li>
    <li>python39Packages.marshmallow-polyfield</li>
    <li>python39Packages.marshmallow-sqlalchemy</li>
    <li>python39Packages.pyatv</li>
    <li>python39Packages.pysiaalarm</li>
    <li>python39Packages.pytenable</li>
    <li>python39Packages.renault-api</li>
    <li>python39Packages.runway-python</li>
    <li>ttp (python39Packages.ttp)</li>
    <li>python39Packages.ttp-templates</li>
    <li>python39Packages.webargs</li>
    <li>sublime-music</li>
    <li>tribler</li>
  </ul>
</details>

File: pkgs/development/python-modules/deepdiff/default.nix [hunk 1]
```diff
@@ -48,10 +49,15 @@ buildPythonPackage rec {
     pyyaml
   ];
 
+  disabledTests = [
+    # Assertion issue with the decimal places
+    "test_get_numeric_types_distance"
```
Comment (ajs124) [inline pkgs/development/python-modules/deepdiff/default.nix:54]: The tests all succeeded for me, why did you disable this one?

**Comment by vcunat** [comment]
Any plans for deepdiff on 21.11?  I expect that [the test failure](https://hydra.nixos.org/build/176849979#tabs-buildsteps) is date-triggered and the same.

**Comment by stasjok** [comment]
A hash for deepdiff is not updated.

**Comment by vcunat** [comment]
Not anymore.  We have now a rebuild due to security updates in some packages.  I'm just about to merge it and thus it would soon propagate to the chanels.

File: pkgs/development/python-modules/deepdiff/default.nix [hunk 2]
```diff
@@ -8,19 +8,20 @@
...
+  disabled = pythonOlder "3.6";
+
   src = fetchFromGitHub {
     owner = "seperman";
     repo = "deepdiff";
-    # 5.7.0 release not tagged https://github.com/seperman/deepdiff/issues/300
-    rev = "f2ffdb83b2993f4f0bb7e854620f0acd0bf6339e";
+    rev = "v${version}";
     hash = "sha256-0UBx7sH2iMrLVl5FtHNTwoecLHi8GbInn75G3FSg4gk=";
   };
```
Comment (vcunat) [inline pkgs/development/python-modules/deepdiff/default.nix:27]: No change in source hash!!  Noticed in https://github.com/NixOS/nixpkgs/pull/173364#issuecomment-1131254806

**Comment by stasjok** [comment]
> Not anymore. We have now a rebuild due to security updates in some packages. I'm just about to merge it and thus it would soon propagate to the channels. EDIT: that's the failure I linked above.

I'm talking about `fetchFromGithub`, where hash is not updated for deepdiff.

File: pkgs/development/python-modules/deepdiff/default.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/development/python-modules/deepdiff/default.nix:54]: So I guess this diff hunk will be the ideal for 21.11: https://github.com/NixOS/nixpkgs/pull/173574

**Comment by vcunat** [comment]
Thanks.  Fix: https://github.com/NixOS/nixpkgs/pull/173575

**Pull request merged**